Tasks/CargoAuthenticateV0/cargoauthenticatemain.ts

```diff
@@ -51,6 +59,45 @@ async function main(): Promise<void> {
             return undefined;
         });
 
+#if WIF
+        const registries = tl.getInput("registryNames");
+        const entraWifServiceConnectionName = tl.getInput("workloadIdentityServiceConnection");
+
+        if (entraWifServiceConnectionName && registries) {
+            const registryNames = registries.split(',');
+            const token = await getFederatedWorkloadIdentityCredentials(entraWifServiceConnectionName);
+
+            if (!token) {
+                throw new Error(tl.loc("FailedToGetServiceConnectionAuth", entraWifServiceConnectionName));
+            }
+
+            for (let registryName of registryNames) {
+                if(!result.registries[registryName]) {
+                    throw new Error(tl.loc("RegistryNotFound", registryName));
+                }
+            }
+
+            const registriesInfo = (<any>Object).entries(result.registries).filter(registry => registryNames.includes(registry[0]));
+            for (let registryInfo of registriesInfo)
+            {
+                var registryUrlStr = url.parse(registryInfo[1].index.replace("sparse+", "")).href;
+                const [registryUrl, tokenName, credProviderName, connectionType] = setRegistryVars(registryUrlStr, registryInfo[0]);
+                if (isValidRegistry(registryUrl, collectionHosts, connectionType))
+                {
+                    tl.warning(tl.loc('ConnectionAlreadySetOverwriting', registryInfo[0], connectionType));
```

Comment:
does this mean we are warning every time we set wif authentication ?

Reply:
Great catch, i'll add a check here

Reply:
If so, I think that its only useful if we are over writting existing authentication. But might not be worth adding extra parsing logic